src/Formik.tsx

```diff
@@ -428,7 +432,7 @@ export class Formik<Values = {}, ExtraProps = {}> extends React.Component<
     return this.runValidations().then(combinedErrors => {
       const isValid = Object.keys(combinedErrors).length === 0;
       if (isValid) {
-        this.executeSubmit();
+        return this.executeSubmit().catch();
```

Comment:
I'm not sure why we catch here. I would expect form submission to reject for cases where form in not valid and submission fails